### ISSUE TITLE
[Dropdown] with icons (with and without label)

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon-addons-iot-react#1373]
### Summary

From 
[Internal Watson-IoT/pal-tracking#175](https://github.ibm.com/Watson-IoT/pal-tracking/issues/175)

> **Is this an extension to an existing component, pattern, or a brand new asset?**
> This is an EDS item.
> 
> **What is the rationale for this contribution.**
> In certain cases, text labels in dropdowns are not able to satisfactorily help users identify what is being referenced in order to make an informed selection, especially when scanning rapidly.
> 
> This contribution aims to support the text values of options in a dropdown with iconography.
> 
> **Detail the intended behavior**
> 
> * This follows all the traditional stylings and behaviours of a traditional carbon dropdown component: https://www.carbondesignsystem.com/components/dropdown/code
> * With the notable exception of a 20px black fill icon left aligned inside the field, and next to each option in the suggestion box.
> 
> Found: https://ibm.invisionapp.com/share/U2NYW8D7DFV#/screens/319194959
> 
> Image:
> ![image](https://media.github.ibm.com/user/684/files/c7626700-200d-11ea-92c2-1e6bc5ed8ed0)
> 
> ![image](https://media.github.ibm.com/user/684/files/c29db300-200d-11ea-91ce-39ccb60b64f0)

**Comment by tay1orjones** [comment]
@lukefirth @bjornalm does the table view dropdown meet this requirement? It looks awfully close. We might be able to consider this resolved, but maybe we need to add the error state?

**Comment by bjornalm** [comment]
It it similar and the view dropdown is highly customisable so it could probably be made to look like this, but the question is if it is worth it. Some differences are label, error state, icon position, checkmark for selected... 

Most of the states described above are probably standard Dropdown stuff, I think it makes more sense to make a new component for this.

**Comment by tay1orjones** [comment]
This item should include the delivery of the inline visual variation, #1374

**Comment by marcJV** [comment]
@dianatran18 
![image](https://user-images.githubusercontent.com/1293444/96169801-510dea00-0ee8-11eb-9768-b04f5d8aacf0.png)

In the invision documentation there is no accessibility text when hovering over the buttons in the dropdown. Should this be included (but not clipped as in the image above), removed, or handled differently?

**Comment by dianatran18** [comment]
> @dianatran18
> ![image](https://user-images.githubusercontent.com/1293444/96169801-510dea00-0ee8-11eb-9768-b04f5d8aacf0.png)
> 
> In the invision documentation there is no accessibility text when hovering over the buttons in the dropdown. Should this be included (but not clipped as in the image above), removed, or handled differently?


Are you referring to these icons in the dropdown?
<img width="224" alt="Screen Shot 2020-10-15 at 4 30 50 PM" src="https://user-images.githubusercontent.com/43549567/96182672-cb983300-0f03-11eb-8827-b532e9baa2e7.png">

I wondered if we even need the option above — I feel like the option with the text next to the icon (example below) is more clear and has less custom pieces to it, but gets the same concept across. I'm not aware of any use cases for the one above as of right now.

<img width="277" alt="Screen Shot 2020-10-15 at 4 36 29 PM" src="https://user-images.githubusercontent.com/43549567/96183218-95a77e80-0f04-11eb-838f-5404452418a8.png">

**Comment by marcJV** [comment]
> Are you referring to these icons in the dropdown?
Yes I am, I can axe that part for now if that is no longer viable.

For the below one: is there a default icon that we need to set for the 'Choose an option' label? Will it be able to be blank?

**Comment by dianatran18** [comment]
That's a good question. I think in this case, we would leave it blank, and the icon would only show once a selection with an icon has been made. The 'Choose an option' text should be left aligned though like a normal dropdown, not leaving a blank icon space. 

Are you following a spec for this? The icons themselves look a little large here. They should be 16px. (I didn't work on this so I'm not too sure how much is available for guidance).

**Comment by marcJV** [comment]
I am going off of the above imagery and the Invision link, is there another place I should be looking at as well?

**Comment by tay1orjones** [comment]
> Are you referring to these icons in the dropdown?
> <img alt="Screen Shot 2020-10-15 at 4 30 50 PM" width="224" src="https://user-images.githubusercontent.com/43549567/96182672-cb983300-0f03-11eb-8827-b532e9baa2e7.png">

@dianatran18 The dropdown with only icon selection buttons is contained in a separate issue, https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1259

**Comment by davidicus** [comment]
> Are you referring to these icons in the dropdown?
> <img alt="Screen Shot 2020-10-15 at 4 30 50 PM" width="224" src="https://user-images.githubusercontent.com/43549567/96182672-cb983300-0f03-11eb-8827-b532e9baa2e7.png">
> 
> I wondered if we even need the option above — I feel like the option with the text next to the icon (example below) is more clear and has less custom pieces to it, but gets the same concept across. I'm not aware of any use cases for the one above as of right now.

https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1259#issuecomment-710392418


Just want to get clarity on the two comments above. Are we dropping this icon grid dropdown? We have 4 products on the issue that have this as a requirement and it has been prioritized as a val 2. 

@dianatran18 @bryancboyd

**Comment by dianatran18** [comment]
I followed up with a few folks to understand their use cases. 

@CynZhang mentioned that the dropdown with the text label (this issue) would be sufficient for their use cases, but @KarenGski said there might be a need for this in the future, to support use cases like a color picker or an icon picker.

If this has already been worked on, I'm fine moving forward with it and see what the adoption for it is. 

I did suggest to scrap the custom part on the bottom, because I actually don't know what it's used for, nor have additional specs for it. 

<img width="276" alt="Screen Shot 2020-10-20 at 9 57 17 AM" src="https://user-images.githubusercontent.com/43549567/96596383-a700da00-12ba-11eb-9bb5-351d776715e7.png">
